### PR TITLE
Refactor Exchange with Discoverer and Trade Order

### LINKED ISSUE #468: Use SDK Discoverer class

### PR DESCRIPTION
**Pull request by Janaka-Steph**
Refactor Exchange page using SDK Discoverer and Trade Order type.
It closes #468 
Based on Louis PR #448 

Please review @tiero @louisinger

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,10 @@
     "serve:no-open": "ionic serve --no-open",
     "sync": "ionic cap sync",
     "tdex:clean": "bash scripts/tdexd-clean.sh",
-    "tdex:run": "curl https://raw.githubusercontent.com/TDex-network/tdex-sdk/master/scripts/tdexd-run -o tdex-run && bash tdex-run && rm tdex-run",
+    "tdex:connect": "docker exec -it tdexd tdexdconnect --out url",
+    "tdex:run": "bash scripts/tdexd-run",
+    "tdex:run-init": "bash scripts/tdexd-run-initialized.sh",
```
Comment (tiero) [inline package.json:58]: whats the difference from this and the `tdex:run-initialized`?

Shall we centralize a script in single place?

File: src/components/ExchangeSearch/index.tsx [hunk 2]
```diff
@@ -35,39 +39,42 @@ const ExchangeSearch: React.FC<ExchangeSearchProps> = ({ prices, assets, setAsse
...
-                  </div>
-                  <div className="search-item-amount">
-                    <span className="price">{(asset.coinGeckoID && prices[asset.coinGeckoID]) || '0'}</span>
-                    <span className="fiat-currency">{currency.toUpperCase()}</span>
-                  </div>
-                </IonItem>
-              );
-            })}
-        </IonList>
+        {assets.length > 0 && (
```
Comment (tiero) [inline src/components/ExchangeSearch/index.tsx:42]: what happens if lenght is 0? Shall we show a text?

File: package.json (hunk 1 above)
Comment (Janaka-Steph) [reply package.json:58]: Running daemon through Docker I need `-v "$(pwd)/tdexd:/home/user/.tdex-daemon"` but CI and you guys needs `-v "$(pwd)/tdexd:/.tdex-daemon"`. So I kept both scripts. I still don't know why we don't have the same config.

File: src/components/ExchangeSearch/index.tsx (hunk 2 above)
Comment (Janaka-Steph) [reply src/components/ExchangeSearch/index.tsx:42]: This is how it looks if no assets
Screenshot 2022-02-18 at 12 55 43

File: package.json (hunk 1 above)
Comment (tiero) [reply package.json:58]: 🤔 🤔 🤔  the docker file should be the same

**Review by tiero — APPROVED** [review]

**Pull request merged**